### ISSUE TITLE
php-phpmd not producing results

### ISSUE THREAD

**Issue by Rimbo** [flycheck/flycheck#757]
Using flycheck 20150921.947, PHPMD 2.2.3.

Instead of errors identified, this pops up in *Messages* --

>Suspicious state from syntax checker php-phpmd: Checker php-phpmd returned non-zero exit code 2, but no errors from output:
```xml
 <?xml version="1.0" encoding="UTF-8" ?>
<pmd version="@project.version@" PHP Warning:  date(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone. in phar:///usr/local/Cellar/phpmd/2.2.3/libexec/phpmd.phar/src/main/php/PHPMD/Renderer/XMLRenderer.php on line 88

>Warning: date(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone. in phar:///usr/local/Cellar/phpmd/2.2.3/libexec/phpmd.phar/src/main/php/PHPMD/Renderer/XMLRenderer.php on line 88
timestamp="2015-09-29T18:44:30+00:00">
  <<<10 gazillion violations deleted --me>>>
</pmd>
```

>Checker definition probably flawed.

**Comment by lunaryorn** [comment]
@Rimbo There's a PHP warning interleaved with the XML output of PHPMD, which makes the XML invalid, causing Flycheck to fail at parsing it.

You need to fix this warning, probably by making some changes to your `php.ini`.  I don't know exactly, because my familiarity with PHP is very superficial.  I'd recommend you to run a Stack Overflow search for this particular warning :smile: 

Closing as `invalid`, because it is not an issue in Flycheck itself, but please feel free to add further comments if you need more help, or think that Flycheck is actually at fault here.

**Comment by Rimbo** [comment]
Thanks -- I was thinking the non-zero return value it was complaining about was the issue. I'll figure out how to make that problem go away. Thanks!
